crates/mipsy_interactive/src/interactive/commands/watchpoint.rs

```diff
@@ -173,6 +173,11 @@ fn watchpoint_insert(state: &mut State, label: &str, args: &[String], remove: bo
             )
         };
 
+        if register == Register::Zero {
+            prompt::error_nl("the zero register cannot be changed, and watching for reads will generate false positives");
```

Comment:
That's between the user and the pseudo-instructions.
If a user want's to watch `$zero` let them watch `$zero`
Plus putting a watch on `$zero` is an easy way of debugging an error like typing `$0` instead of `$t0` (just ask Andrew)

Reply:
You also allow watching of `$at` and `$k*` which would have the same issues as you have described for `$zero`.
I just don't see a point putting a restriction in place. If it's not useful then the user can just not use it.

Reply:
`$zero` is unique in that it will produce actual false positives, not just positives arising from the expansion of pseudoinstructions due to how I parse the affected registers. For example, it will detect `mult $t0, $t1` as reading from `$zero` due to the layout of the machine code

Reply:
`mips.yaml` already has a `reads` field that you can use.
![image](https://user-images.githubusercontent.com/22044664/191413568-a6e67806-df75-42df-832b-2382f93ae51b.png)

Reply:
Yeah I considered it initially but figured it would make more sense to just parse the machine code according to the MIPS specification - very similar things are done in mipsy_lib anyway. The only quirk I'm aware of is this zero register thing, but it doesn't affect the functional correctness of anything except putting a watchpoint on reads of the zero register which I didn't consider a dealbreaker

Reply:
Using this method will probably also make watching on memory locations easier compared to mips.yaml